src/ebonite/ext/sqlalchemy/repository.py

```diff
@@ -116,13 +118,18 @@ def _create_object(self, object_type: Type[Attaching], obj: T, error_type) -> T:
             obj._id = p.id
             return obj
 
-    def _delete_object(self, object_type: Type[Attaching], obj, error_type):
+    def _delete_object(self, object_type: Type[Attaching], obj, ne_error_type, ie_error_type):
         with self._session() as s:
             p = s.query(object_type).filter(object_type.id == obj.id).first()
             if p is None:
-                raise error_type(obj)
+                raise ne_error_type(obj)
             logger.debug('Deleting object %s', p)
-            s.delete(p)
+            try:
+                s.delete(p)
+                s.commit()
+            except IntegrityError:
+                s.rollback()
```

Comment:
`_session` context manager guarantees `rollback` on exception. Please take a look at it and `_create_object` as reference.